l10n_fr_hr_ssnid_validation/models/hr_employee.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from stdnum import get_cc_module
+from stdnum.exceptions import (
+    InvalidChecksum,
+    InvalidComponent,
+    InvalidFormat,
+    InvalidLength,
+)
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    ssnid_key = fields.Char(
+        string='SSN No Key',
+        help='Social Security Number Key',
+        size=2,
+        required=True,
+    )
+
+    @api.onchange('ssnid', 'ssnid_key')
+    def onchange_ssnid(self):
+        self.check_ssnid()
+
+    @api.constrains('ssnid', 'ssnid_key')
+    def check_ssnid(self):
+        # We assume this dev work only in France
+        # other available format are on
+        # https://arthurdejong.org/python-stdnum/doc/1.13/index.html#available-formats
+        mod = get_cc_module('fr', 'nir')
```

Comment:
can you rely on company employee belongs to? it will prevent errors for multicompany mode